contrib/guix/guix-clean

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env bash
+export LC_ALL=C
+set -e -o pipefail
+
+# Source the common prelude, which:
+#   1. Checks if we're at the top directory of the Monero repository
+#   2. Defines a few common functions and variables
+#
+# shellcheck source=libexec/prelude.bash
+source "$(dirname "${BASH_SOURCE[0]}")/libexec/prelude.bash"
+
+
+###################
+## Sanity Checks ##
+###################
+
+################
+# Required non-builtin commands should be invokable
+################
+
+check_tools cat mkdir make git guix
+
+
+#############
+##  Clean  ##
+#############
+
+# Usage: under_dir MAYBE_PARENT MAYBE_CHILD
+#
+# If MAYBE_CHILD is a subdirectory of MAYBE_PARENT, print the relative path
+# from MAYBE_PARENT to MAYBE_CHILD. Otherwise, return 1 as the error code.
+#
+# NOTE: This does not perform any symlink-resolving or path canonicalization.
+#
+under_dir() {
+    local path_residue
+    path_residue="${2##"${1}"}"
```

Comment:
This function is easy to get wrong and could use a test suite: For safety, you'd want to check that the child starts with "$parent/". Otherwise, as it stands, you get results such as:
```
$ under_dir /tmp /tmpx/config.py  
x/config.py
```

Also, while this function name is clear enough in the local context, if you end up making it a library function, I'd consider something such as `path_contains`, to emphasize that it's just testing path-based containment, not actual containment. As for testing, you could just put this function in its own file, then source it here and via the test (dumping it into one huge library has the disadvantage that, as the library grows, the set of false implicit dependencies grows, which makes scripts more bloated).

Reply:
One other thing: Normally it's helpful to, in addition to adding a "/" to parent, support the case where parent already contains one (pretty easy to do by e.g. dropping all trailing slashes and then adding one).

Reply:
I fixed the edge cases in this function. Since the function is only used here, I didn't want to move it.

I believe I have addresses all your comments.